src/store/reducers/topic.ts

```diff
@@ -142,4 +151,53 @@ export const selectPreparedConsumersData: Selector<
     });
 });
 
+export const selectPreparedTopicPartitions: Selector<
+    ITopicRootStateSlice,
+    PreparedPartitionData[] | undefined
+> = createSelector([selectPartitions], (partitions) => {
+    return partitions?.map((partition) => {
+        // describe_topic endpoint doesn't return zero values, so some values will be initialized with 0
+        const {partition_id = '0', partition_stats} = partition;
+
+        const {
+            partition_offsets,
+            store_size_bytes = '0',
+            last_write_time: partition_last_write_time,
+            max_write_time_lag: partition_write_lag,
+            bytes_written,
+            partition_node_id = 0,
+        } = partition_stats || {};
+
+        const {start: start_offset = '0', end: end_offset = '0'} = partition_offsets || {};
+
+        return {
+            partitionId: partition_id,
+            storeSize: store_size_bytes,
+
+            writeSpeed: convertBytesObjectToSpeed(bytes_written),
+            readSpeed: undefined,
+
+            partitionWriteLag: parseLag(partition_write_lag),
+            partitionWriteIdleTime: parseTimestampToIdleTime(partition_last_write_time),
+
+            consumerWriteLag: undefined,
+            consumerReadLag: undefined,
+            consumerReadIdleTime: undefined,
+
+            uncommitedMessages: undefined,
+            unreadMessages: undefined,
+
+            startOffset: start_offset,
+            endOffset: end_offset,
+            commitedOffset: undefined,
+
+            readSessionId: undefined,
```

Comment:
Could you please clarify if these properties may be not undefined?

Reply:
They are always `undefined` for partitions from `/describe_topic` endpoint. And always defined for `/describe_consumer`